### DISCUSSION TITLE
I have problem while apply cam on my own custom model, should I change my model to resnet?

### DISCUSSION THREAD

**Discussion by JohnnyM1ng** [frgfm/torch-cam#238] [Q&A] [answered]
Hi torch-cam provider XD
I want to apply cam to my own model(a SE-ResNet which has 6 input channels instead of 3 in ResNet).But I met some problems when I follow your example.
`model = se_resnet18(6, 2).eval()
cam_extractor = SmoothGradCAMpp(model)`
And a RuntimeError shows up like this
`RuntimeError: Given groups=1, weight of size [64, 6, 7, 7], expected input[1, 3, 224, 224] to have 6 channels, but got 3 channels instead`
I have read one similar discussion #https://github.com/frgfm/torch-cam/discussions/135,but it seems can not solve my problem.Have I missed anything?
Thank you sincerely.

**Comment by JohnnyM1ng** [comment, marked as answer]
Ok，I have figured out what is wrong, I should set the input_shape XD
`cam_extractor = SmoothGradCAMpp(model, input_shape=(6, 64, 64))`

**Reply by frgfm** [reply]
Hey @JohnnyM1ng :wave: 

Yup exactly, starting to think I should add this as a suggestion in the error message when the shapes don't match haha
Glad you found the answer either way!
